Azure/azure-cli
# [Profile] `az login`: Fall back to device code flow in GitHub Codespaces

**Related command**

`az login`

**Description**

This PR fixes the long standing issue of `az login` not working in GitHub Codespaces (https://github.com/Azure/azure-cli/issues/20315).
For that it uses a simple workaround, the same used in the gh CLI: if the `CODESPACES=true` environnement variable is set, it falls back to using a device code login.

**Testing Guide**

1. In a bash terminal: run `az login` -> by default, it should run the web browser login
2. Run `CODESPACES=true az login` -> it should automatically fallback to device code login. You can also simply run `az login` in GitHub Codespaces :)

**History Notes**


[Core] `az login`: Add fallback to device code in GitHub Codespaces (#20315)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
